tests/chaos/utils.py

```
@@ -415,3 +448,84 @@ def get_instance_type(name):
+        namespace=namespace_name,
+        get_all=True,
+    )
+
+    if not pods:
+        LOGGER.error("No pods found by prefix — this is unexpected")
+        return
+
+    delete_count = max(1, round(ratio * len(pods)))
+    pods_to_delete = random.sample(pods, min(delete_count, len(pods)))
+
+    LOGGER.info(f"Deleting {len(pods_to_delete)} out of {len(pods)} pods")
```

Ruff (0.14.14)

[warning] Logging statement uses f-string

(G004)